### DISCUSSION TITLE
Bug: Eslint stopped working in VSCode and Vite / React project

### DISCUSSION THREAD

**Discussion by JulSeb42** [eslint/eslint#18673] [Q&A]
### Environment

Node version: 21.2
npm version: 10.2.4
Local ESLint version: 8.57.0 & 9.x
Global ESLint version: -
Operating System: macOS 14.4.1
Vite: 5.3.1
vite-plugin-eslint: 1.8.1
Typescript: 5.5.3
React: 18.3.1

### What parser are you using?

@typescript-eslint/parser

### What did you do?

I ran `npm init @eslint/config@latest` inside a Vite / React project, selected

```
? How would you like to use ESLint? …
  To check syntax only
❯ To check syntax and find problems

What type of modules does your project use? …
❯ JavaScript modules (import/export)
  CommonJS (require/exports)
  None of these

Which framework does your project use? …
❯ React
  Vue.js
  None of these

The React plugin doesn't officially support ESLint v9 yet. What would you like to do? …
❯ Install ESLint v9.x with compatibility utilities
  Install ESLint v8.x

Does your project use TypeScript? …
  No
❯ Yes

Where does your code run? …  (Press <space> to select, <a> to toggle all, <i> to invert selection)
✔ Browser
• Node

The config that you've selected requires the following dependencies:
eslint@9.x, globals, @eslint/js, typescript-eslint, eslint-plugin-react, @eslint/compat
? Would you like to install them now? › No / **Yes**

Which package manager do you want to use? …
  npm
❯ yarn
  pnpm
  bun
```

Install went well.  Then added this rule in the new `eslint.config.js`:
```
rules: {
    "no-duplicate-imports": "warn",
}
```

Added this at the top of my `App.tsx`:
```
import { useState } from "react"
import { useEffect } from "react"
```

The warning shows in the terminal, but is not shown in VSCode:

```shell
warning  'useEffect' is defined but never used
```

<img width="237" alt="Capture d’écran 2024-07-08 à 18 55 33" src="https://github.com/eslint/eslint/assets/23702761/5bcba4f8-0cba-42be-82ec-9e30a9ec425c">

### What did you expect to happen?

Get a warning in VSCode, where the problem is underlined in yellow

### What actually happened?

Nothing happened

### Link to Minimal Reproducible Example

https://github.com/JulSeb42/eslint-bug

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by aladdin-add** [comment]
In versions of vscode-eslint prior to v3.0.10, the new configuration system is not enabled by default. To enable support for the new configuration files, edit your .vscode/settings.json file and add the following:

```json
{
  // required in vscode-eslint < v3.0.10 only
  "eslint.experimental.useFlatConfig": true
}
```
https://eslint.org/docs/latest/use/configure/migration-guide#visual-studio-code-support
